cmake/test_runner.py

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python
+import inspect
+import os
+import sys
+
+import imp
+
+try:
+    from unittest2 import TextTestRunner, TestLoader, TestCase
+except ImportError:
+    from unittest import TextTestRunner, TestLoader, TestCase
+
+
+def runTestCase(tests, verbosity=0):
+    test_result = TextTestRunner(verbosity=verbosity).run(tests)
+
+    if len(test_result.errors) or len(test_result.failures):
+        test_result.printErrors()
+        return False
+    else:
+        return True
```

Comment:
If you grab onto the `runner`, then you can query `runner.wasSuccessful()` instead of checking length of stuff.